src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/dagligReise/beregning/OffetnligTransportBeregningServiceUtil.kt

```diff
@@ -0,0 +1,153 @@
+        .sumOf { it.antallDager }
+
+fun finnBilligsteAlternativForTrettidagersPeriode(grunnlag: Beregningsgrunnlag): Int =
+    min(finnBilligsteKombinasjonAvEnkeltBilettOgSyvdagersBillett(grunnlag), grunnlag.pris30dagersbillett)
+
+/**
+ * Minimum Cost For Tickets.
+ * Doc: https://docs.vultr.com/problem-set/minimum-cost-for-tickets
+ */
+private fun finnBilligsteKombinasjonAvEnkeltBilettOgSyvdagersBillett(grunnlag: Beregningsgrunnlag): Int {
+    val resiedagerPerUke = finnReisedagerPerUke(grunnlag)
+    val reisedagerListe = lagReisedagerListe(resiedagerPerUke)
+
+    // Hvis ingen reisedager er billigste pris 0kr
+    if (reisedagerListe.isEmpty()) return 0
+
+    val sisteReiseDag = reisedagerListe.last()
+    val reisekostnader = MutableList(sisteReiseDag + 1) { 0 }
+
+    var reisedagIndeks = 0
+    for (gjeldeneDag in 1..sisteReiseDag) {
+        if (gjeldeneDag.skalIkkeReise(reisedagerListe, reisedagIndeks)) {
+            reisekostnader[gjeldeneDag] = reisekostnader[gjeldeneDag - 1]
+        } else {
+            reisedagIndeks++
+            reisekostnader[gjeldeneDag] =
+                listOfNotNull(
+                    finnReisekostnadForNyEnkeltbillett(gjeldeneDag, reisekostnader, grunnlag),
+                    finnReisekostnadForNySyvdagersbillett(gjeldeneDag, reisekostnader, grunnlag),
+                ).min()
+        }
+    }
+    return reisekostnader[sisteReiseDag]
+}
+
+private fun finnReisedagerPerUke(grunnlag: Beregningsgrunnlag): Map<Uke, PeriodeMedDager> =
+    grunnlag.splitPerUke { fom, tom ->
+        val uke = Datoperiode(fom, tom)
+        val dagerIPerioden = finnAntallDagerISnittetMellomUkeOgVedtaksperioder(uke, grunnlag.vedtaksperioder)
+        min(grunnlag.antallReisedagerPerUke, dagerIPerioden)
+    }
+
+/**
+ * Retunerer en liste med dager hvor brukeren skal reise.
+ * Dagene telles fra første dagen i perioden.
+ * E.g hvis brukeren skal reise man-ons i to uker så skal brukeren reise dag 1,2,3,8,9,10
+ * og listen vil se slik ut: [1,2,3,8,9,10]
+ */
+private fun lagReisedagerListe(reisedagerPerUke: Map<Uke, PeriodeMedDager>): List<Int> =
+    reisedagerPerUke.values.flatMapIndexed { indeks, periodeMedDager ->
+        val startDag = reisedagerPerUke.firstNotNullOf { it.value.fom }
+        val dagerTilMandag = finnDagerTilMandagIUke(startDag, indeks)
+        // dag + 1 for å 1-indeksere tellingen av dager
+        val list = List(periodeMedDager.antallDager) { dag -> dagerTilMandag + dag + 1 }
+        list
+    }
+
+/**
+ * Retunerer 0 for `ukeNr` = 0,
+ * derreter antall dager fra `startDag` til mandag i `ukeNr`
+ */
+private fun finnDagerTilMandagIUke(
+    startDag: LocalDate,
+    ukeNr: Int,
+): Int {
+    val nesteMandag = startDag.with(TemporalAdjusters.next(DayOfWeek.MONDAY))
+    val dagerTilMandag = ChronoUnit.DAYS.between(startDag, nesteMandag).toInt()
+    return when (ukeNr) {
+        0 -> 0
+        1 -> dagerTilMandag
+        else -> (ukeNr - 1) * 7 + dagerTilMandag
+    }
+}
```

Comment:
Kan eventuelt bruke bare de innebygde tidsfunksjonene i Java, om vi syntes det er litt enklere å forstå 
```suggestion
/**
 * Beregner antall dager fra startDato til mandagen i en gitt uke (ukeNr).
 * For den første uken (ukeNr = 0) returneres 0.
 * For påfølgende uker beregnes antall dager til mandagen i den gitte uken.
 */
private fun tellDagerTilStartenAvUke(
    startDag: LocalDate,
    ukeNr: Int,
): Int {
    if (ukeNr == 0) return 0
    val mandagenIGittUke = startDag.with(TemporalAdjusters.next(DayOfWeek.MONDAY)).plusWeeks(ukeNr - 1L)
    return ChronoUnit.DAYS.between(startDag, mandagenIGittUke).toInt()
}
```